pkg/routeagent/controllers/route/driver.go

```diff
@@ -44,12 +41,12 @@ func discoverCNIInterface(clusterCIDR string) *cniInterface {
 				if clusterNetwork.Contains(address) {
 					klog.V(log.DEBUG).Infof("Found CNI Interface %q that has IP %q from ClusterCIDR %q",
 						iface.Name, ipAddr.String(), clusterCIDR)
-					return &cniInterface{ipAddress: ipAddr.String(), name: iface.Name}
+					return &cniInterface{ipAddress: ipAddr.String(), name: iface.Name}, nil
 				}
 			}
 		}
 	}
-	return nil
+	return nil, fmt.Errorf("unable to find CNI Interface on the host which has IP from %q", clusterCIDR)
```

Comment:
I'm afraid no. This is an unexpected case. We ideally expect to discover the CNIInterface on the host. In the rare case, if it's missing, we want to treat that as an error (instead of PANIC) and continue to function with some missing functionality.